### ISSUE TITLE
Policies' Client Roles are not honored (or, I don't get it 🤦)? V2.0.0-pre-3 on .Net 8 Web Api

### ISSUE THREAD

**Issue by TheAutomaTom** [NikiforovAll/keycloak-authorization-services-dotnet#82]
I intended to only use Client Roles for an app.  I can see the client roles in the token, but they are not honored by my Api.  Realm level roles work as expected.  I also tried a combination, and still the presence of a Client Role requirement prevents authorization.

```cs
// Program.cs
 builder.Services.AddKeycloakWebApiAuthentication(configuration, o =>
      {
        if (env == "Development" || env == "Test") // Disable Auth's Https requirement.
        {
          o.RequireHttpsMetadata = false;

        }
        //options.Audience = configuration.GetSection("Keycloak:Audience").Value; // Is an audience required?
      });
      
      builder.Services.AddAuthorization(options =>
      {
        options.AddPolicy("Reader", p =>
        {
          // p.RequireRealmRoles("RealmReader"); // Is a realm role always required?
          p.RequireResourceRoles("Reader");
          // p.RequireRole("Reader"); // Is this redundant?  It's in the samples.
        });
      })
      .AddKeycloakAuthorization();
```

```cs
//Controller
    [HttpGet, Authorize(Policy ="Reader")]
    public IActionResult ReadersOnly()
    {
      return Ok($"{nameof(ReadersOnly)} worked!");
    }
```

```js
// Decoded jwt
3bd6d0481312",
  "acr": "1",
  "allowed-origins": [
    "/*"
  ],
  "realm_access": {
    "roles": [
      "default-roles-dev-lab",
      "offline_access",
      "RealmReader",   // <==== This works if used alone.
      "uma_authorization"
    ]
  },
  "resource_access": {
    "dev-client": {
      "roles": [
        "Reader"  // <==== Expected requirement.
      ]
    },
    "account": {
      "roles": [
        "manage-account",
        "manage-account-links",
        "view-profile"
      ]
    }
  },
```

Thanks for providing a great looking product and all the support! Tom G.

**Comment by NikiforovAll** [comment]
Could you please show the configuration? Please feel free to contact me alexey.nikiforovall@gmail.com so we can schedule a call to discuss your issue, I would like it to be resolved before release of 2.0.0.

Please consult integration tests for correct setup: https://github.com/NikiforovAll/keycloak-authorization-services-dotnet/blob/332376d50cb8c0f4533247e89bf42fe457695837/tests/Keycloak.AuthServices.IntegrationTests/PolicyTests.cs#L76

**Comment by NikiforovAll** [comment]
I found the issue, we need to provide a configuration to `.AddKeycloakAuthorization(options => {options.Resource = "dev-client"});`. I will add method overloads that accept configuration and configuration section. Basically, we need to tell the client name as role source. 

Basically, all components depend on their own configurations, therefore we need to configure them separately. Authorization doesn't depend on Authentication and vice versa.

**Comment by TheAutomaTom** [comment]
Thanks for responding so quickly, I will see if I can implement that fix today!

**Comment by NikiforovAll** [comment]
I'm working on adding method overload that accepts `IConfiguration`, meantime you can use the workaround I've provided.

**Comment by NikiforovAll** [comment]
Should be fixed in 2.0.0, please feel free to reopen if you encounter any isses.
